### PR TITLE
feat: enable admin to delete any order

### LINKED ISSUE #6173: Allow admin to soft delete orders
To include an order table in admin section, the admin needs to  have access to delete any order

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #6173 

#### Short description of what this resolves:
Allow admin to soft delete any order

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
And what happens on deleting order? Refund?

**Comment by uds5501** [comment]
@iamareebjamal No, the project has no refund ability yet.
This PR is just to ensure that `admin` can delete an order in any case, just the way a `co-organizer` can.

**Comment by iamareebjamal** [comment]
But I see no changes related to admin

**Comment by uds5501** [comment]
@iamareebjamal sorry for that, forgot to change admin permissions.
Apart from that, i've added the changes.

**Comment by uds5501** [comment]
@iamareebjamal 
- Since refund feature has not yet been implemented, should the admin be allowed to delete completed and placed orders?

- @CosmicCoder96 I tried deleting few orders and observing their behaviour, they go through `before_update` hook. I am trying to ensure soft deletion so that the orders can be restored. (just like deleting and restoring events)

File: app/api/orders.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def before_update_object(self, order, data, view_kwargs):
                 # Order created from the public pages.
                 for element in data:
                     if data[element] and data[element] != getattr(order, element, None):
-                        if element not in ['status', 'deleted_at']:
+                        if element != 'status' and element != 'deleted_at':
```
Comment (iamareebjamal) [inline app/api/orders.py]: It was better before IMO

Comment (uds5501) [reply app/api/orders.py]: @iamareebjamal arlgiht, changing it.

**Comment by uds5501** [comment]
@iamareebjamal @shreyanshdwivedi squashed the commits and resolved the conflicts, please review now.

File: app/api/orders.py [hunk 2]
```diff
@@ -314,37 +314,31 @@ def before_update_object(self, order, data, view_kwargs):
...
-                                raise ForbiddenException({'pointer': 'data/status'},
-                                                         "You cannot update the status of a completed paid order")
-                            elif element == 'status' and order.status == 'cancelled':
-                                # Since the tickets have been unlocked and we can't revert it.
-                                raise ForbiddenException({'pointer': 'data/status'},
-                                                         "You cannot update the status of a cancelled order")
-                        else:
-                            check_event_user_ticket_holders(order, data, element)
+                    if data[element] and data[element] != getattr(order, element, None):
+                        if element not in ['status', 'deleted_at']:
```
Comment (mrsaicharan1) [inline app/api/orders.py]: let's push this to the previous line instead of additional nesting.

**Comment by kushthedude** [comment]
@uds5501 Are we good closing it ?

**Comment by uds5501** [comment]
@kushthedude Yes. Not a priority right now.

**Pull request closed without merging**